sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { CredentialUnavailableError } from "../client/errors";
+import { credentialLogger, formatSuccess, formatError } from "../util/logging";
+import { trace } from "../util/tracing";
+import { ensureValidScope, getScopeResource } from "../util/scopeUtils";
+import { processUtils } from "../util/processUtils";
+
+const logger = credentialLogger("AzurePowerShellCredential");
+
+/**
+ * Formats a command based on the platform executing it.
+ * @internal
+ */
+export function formatCommand(commandName: string): string {
+  if (process.platform === "win32") {
+    return `${commandName}.exe`;
+  } else {
+    return commandName;
+  }
+}
+
+/**
+ * Receives a list of commands to run, executes them, then returns the outputs.
+ * If anything fails, an error is thrown.
+ * @internal
+ */
+async function runCommands(commands: string[][]): Promise<string[]> {
+  const results: string[] = [];
+
+  try {
+    for (const command of commands) {
+      const [file, ...parameters] = command;
+      const buffer = await processUtils.execFile(file, parameters);
+      results.push(buffer.toString("utf-8"));
+    }
+  } catch (e) {
+    throw new Error(e.toString("utf-8"));
```

Comment:
maybe it would be better to have the execFile helper throw new Error on stderr so you don't have to convert the buffer here to another Error object? That way you won't re-wrap the original error if there's an i/o error or something from the operation

Reply:
can do!

Reply:
Done!